src/PlusDataCollection/WinProbe/vtkPlusWinProbeVideoSource.cxx

```diff
@@ -749,6 +749,8 @@ PlusStatus vtkPlusWinProbeVideoSource::InternalConnect()
   {
     LOG_WARNING("Failed setting session pointer!");
     WPDisconnect();
+    WPDispose();
```

Comment:
Maybe it is better to put these into `~vtkPlusWinProbeVideoSource` (class's destructor)?

Reply:
I tested just the new calls in the destructor as seen below, but it doesn't fix the disconnect problem.
```c++
vtkPlusWinProbeVideoSource::~vtkPlusWinProbeVideoSource()
{
  if(this->Connected)
  {
    this->Disconnect();
    WPDispose();
    WPDXDispose();
  }
}
```

Reply:
`WPDispose();` and `WPDXDispose();` should not be within the `if` clause. If that doesn't help, then your original commit is good to go.

Reply:
In the destructor, the device should already be disconnected.

Reply:
See latest change. Putting it outside of that `if` clause does work.